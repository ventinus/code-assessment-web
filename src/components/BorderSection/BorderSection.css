.border-section {
  border: 0 solid #d8d8d8; }
