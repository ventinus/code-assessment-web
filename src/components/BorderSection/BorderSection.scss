@import '../../scss/variables/mixins.scss';
@import '../../scss/variables/variables.scss';

.border-section {
  border: 0 solid $grey-4;
}
