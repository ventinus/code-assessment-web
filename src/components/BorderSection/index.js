import React from 'react'
import PropTypes from 'prop-types'
import { capitalize } from '../../helpers'

import './BorderSection.css'

const BorderSection = ({ tag = 'div', side = 'bottom', style = {}, className, children }) => (
  React.createElement(tag, {
    style: {
      ...style,
      [`border${capitalize(side)}Width`]: 1
    },
    className: `border-section ${className ? className : ''}`
  }, children)
)

BorderSection.propTypes = {
  tag: PropTypes.string,
  side: PropTypes.string,
  style: PropTypes.object,
  className: PropTypes.string,
  children: PropTypes.any,
}

export default BorderSection
