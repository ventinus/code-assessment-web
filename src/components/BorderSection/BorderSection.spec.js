import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import BorderSection from '.'

const setup = props => shallow(<BorderSection {...props} />)

describe('BorderSection component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<BorderSection>children</BorderSection>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders the appropriate tag', () => {
    const component = setup({tag: 'li'})
    expect(component.name()).toBe('li')
  })

  it('renders the correct border style', () => {
    const component = setup({side: 'top'})
    expect(component.props().style).toEqual({borderTopWidth: 1})
  })

  it('renders the custom className as an addition', () => {
    const component = setup({className: 'special'})
    expect(component.props().className).toBe('border-section special')
  })
})
