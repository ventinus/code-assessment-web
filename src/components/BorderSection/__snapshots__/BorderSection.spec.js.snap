// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BorderSection component renders correctly 1`] = `
<div
  className="border-section "
  style={
    Object {
      "borderBottomWidth": 1,
    }
  }
>
  children
</div>
`;
